### DISCUSSION TITLE
As SQLite is getting traction, move database.sqlite to storage folder

### DISCUSSION THREAD

**Discussion by RiadhKHEDHIRI** [laravel/framework#54017] [Ideas]
Hi Community,

I was playing around with Laravel, SQLite, Forge, and Envoyer.

I noticed that the database gets wiped when deploying because it is not in _special_ folders such as `storage`.

The workaround I found is to alter `config/database.php` from

`'database' => env('DB_DATABASE', database_path('db/database.sqlite')),` to

`'database' => env('DB_DATABASE', storage_path('db/database.sqlite')),` as we cannot specify a relative path in the `.env` file.

I thought of something like this `storage/database/{{APP_ENV}}/database.sqlite`.

I don't know if this is a long-term solution.

As SQLite is getting traction via Turso, the default RoR setup switched to it to make launching easier, faster, and cheaper by avoiding PaaS. Thus, having the same thing in the Laravel Community would be nice.

What do you think? Thank you in advance.

**Comment by Sophist-UK** [comment]
Yes - I have also moved my database to storage/database.

However:

1. Any improvement needs to be backwards compatible for existing files.
2. I think that "database.sqlite" should be "laravel.sqlite" in the same way that the default mysql database is called laravel.
3. I think that the APP_ENV should be part of the filename.

```php
'database' => env('DB_DATABASE', 
    file_exists(database_path('database.sqlite') ? database_path('database.sqlite') : storage_path('database/laravel_' . ENV('APP_ENV', '') . '.sqlite')),
```
